packages/expo-device/ios/EXDevice/EXDevice.m

```diff
@@ -0,0 +1,256 @@
+    }];
+  } else {
+    resolve(UMNullIfNil(strongSelf.webViewUserAgent));
+  }
+}
+
+UM_EXPORT_METHOD_AS(getCarrierAsync,
+                    getCarrierAsyncWithResolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  resolve(self.carrier ?: [NSNull null]);
+}
+
+UM_EXPORT_METHOD_AS(getMACAddressAsync,
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject) {
+  //some iOS privacy issues
+  NSString *address = @"02:00:00:00:00:00";
+  resolve(address);
+}
+
+UM_EXPORT_METHOD_AS(getIpAddressAsync,
+                    getIpAddressAsyncWithResolver:(UMPromiseResolveBlock)resolve rejecter:(UMPromiseRejectBlock)reject)
+{
+  NSString *address = @"0.0.0.0";
+  struct ifaddrs *interfaces = NULL;
+  struct ifaddrs *temp_addr = NULL;
+  int success = 0;
+  // retrieve the current interfaces - returns 0 on success
+  success = getifaddrs(&interfaces);
+  
+  //reject if we fail to retrieve any interfaces
+  if (success != 0) {
+    // Free memory
+    freeifaddrs(interfaces);
+    
+    reject(@"E_NO_IFADDRS", @"No network interfaces could be retrieved.", nil);
+    return;
+  }
+  
+  else if (success == 0) {
+    // Loop through linked list of interfaces
+    temp_addr = interfaces;
+    while(temp_addr != NULL) {
+      if(temp_addr->ifa_addr->sa_family == AF_INET) {
+        // Check if interface is en0 which is the wifi connection on the iPhone
+        if([[NSString stringWithUTF8String:temp_addr->ifa_name] isEqualToString:@"en0"]) {
+          // Get NSString from C String
+          address = [NSString stringWithUTF8String:inet_ntoa(((struct sockaddr_in *)temp_addr->ifa_addr)->sin_addr)];
+        }
+      }
+      temp_addr = temp_addr->ifa_next;
+    }
+  }
+  // Free memory
+  freeifaddrs(interfaces);
+  resolve(address);
+}
+
+UM_EXPORT_METHOD_AS(isPinOrFingerprintSetAsync, isPinOrFingerprintSetAsyncWithResolver:(UMPromiseResolveBlock)resolve rejecter:(UMPromiseRejectBlock)reject)
+{
+#if TARGET_OS_TV
+  BOOL isPinOrFingerprintSet = NO;
+#else
+  LAContext *context = [[LAContext alloc] init];
+  BOOL isPinOrFingerprintSet = ([context canEvaluatePolicy:LAPolicyDeviceOwnerAuthentication error:nil]);
+#endif
+  resolve(@(isPinOrFingerprintSet));
+}
+
+UM_EXPORT_METHOD_AS(getFreeDiskStorageAsync, getFreeDiskStorageAsyncWithResolver:(UMPromiseResolveBlock)resolve rejecter:(UMPromiseRejectBlock)reject)
+{
+  resolve([self freeDiskStorage]);
+}
+
+- (NSString *)deviceId
+{
+  struct utsname systemInfo;
+  
+  uname(&systemInfo);
+  
+  NSString* deviceId = [NSString stringWithCString:systemInfo.machine
+                                          encoding:NSUTF8StringEncoding];
+  
+  if ([deviceId isEqualToString:@"i386"] || [deviceId isEqualToString:@"x86_64"] ) {
+    deviceId = [NSString stringWithFormat:@"%s", getenv("SIMULATOR_MODEL_IDENTIFIER")];
+  }
+  
+  return deviceId;
+}
+
+- (NSString *)carrier
+{
+#if (TARGET_OS_TV)
```

Comment:
Remove for now (we'll need to add something similar back when we support Catalyst for Mac)